source/android/mobilechatapp/src/main/java/com/example/mobilechatapp/MainActivity.java

```diff
@@ -263,22 +263,32 @@ public void run()
         this.runOnUiThread(new RunnableExtended(this, text, duration));
     }
 
-    private void onSubmit(BaseActionElement actionElement, RenderedAdaptiveCard renderedAdaptiveCard) {
+    private void onSubmit(BaseActionElement actionElement, RenderedAdaptiveCard renderedAdaptiveCard)
+    {
         SubmitAction submitAction = null;
-        if (actionElement instanceof SubmitAction) {
+        if (actionElement instanceof SubmitAction)
+        {
             submitAction = (SubmitAction) actionElement;
-        } else if ((submitAction = SubmitAction.dynamic_cast(actionElement)) == null) {
+        }
+        else if ((submitAction = SubmitAction.dynamic_cast(actionElement)) == null)
+        {
             throw new InternalError("Unable to convert BaseActionElement to ShowCardAction object model.");
         }
 
         String data = submitAction.GetDataJson();
         Map<String, String> keyValueMap = renderedAdaptiveCard.getInputs();
         if (!data.isEmpty())
         {
-            try {
-                JSONObject object = new JSONObject(data);
+            try
+            {
+                JSONObject object = new JSONObject();
+                if (!data.equals("null\n"))
```

Comment:
>null\n [](start = 34, length = 6)

Where is this string coming from? Is getting this string the expected behavior or should the shared model be doing something better? If we  have to do this, third parties would do, and it seems like a bad experience.

Reply:
that string is coming from the shared model